src/main/java/ch/njol/skript/expressions/ExprCommandInfo.java

```diff
@@ -43,108 +47,105 @@
+		ALIASES(null), // Handled differently
+		PERMISSION(Command::getPermission),
+		PERMISSION_MESSAGE(Command::getPermissionMessage),
+		PLUGIN(command -> {
+			if (command instanceof PluginCommand) {
+				return ((PluginCommand) command).getPlugin().getName();
+			} else if (command instanceof BukkitCommand) {
+				return "Bukkit";
+			} else if (command.getClass().getPackage().getName().startsWith("org.spigot")) {
+				return "Spigot";
+			} else if (command.getClass().getPackage().getName().startsWith("com.destroystokyo.paper")) {
+				return "Paper";
+			}
+			return "Unknown";
+		});
+
+		private final Function<Command, String> function;
+
+		InfoType(Function<Command, String> function) {
+			this.function = function;
+		}
+
 	}
 
 	static {
-		Skript.registerExpression(ExprCommandInfo.class, String.class, ExpressionType.SIMPLE,
-			"[the] main command [label] of command %strings%", "command %strings%'[s] main command [name]",
-			"[the] description of command %strings%", "command %strings%'[s] description",
-			"[the] label of command %strings%", "command %strings%'[s] label",
-			"[the] usage of command %strings%", "command %strings%'[s] usage",
-			"[(all|the|all [of] the)] aliases of command %strings%", "command %strings%'[s] aliases",
-			"[the] permission of command %strings%", "command %strings%'[s] permission",
-			"[the] permission message of command %strings%", "command %strings%'[s] permission message",
-			"[the] plugin [owner] of command %strings%", "command %strings%'[s] plugin [owner]");
+		Skript.registerExpression(ExprCommandInfo.class, String.class, ExpressionType.PROPERTY,
+			"[the] main command [label] [of [[the] command[s] %-strings%]]", "command[s] %strings%'[s] main command [name]",
+			"[the] description [of [[the] command[s] %-strings%]]", "command[s] %strings%'[s] description",
+			"[the] label [of [[the] command[s] %-strings%]]", "command[s] %strings%'[s] label",
+			"[the] usage [of [[the] command[s] %-strings%]]", "command[s] %strings%'[s] usage",
+			"[(all|the|all [of] the)] aliases [of [[the] command[s] %-strings%]]", "command[s] %strings%'[s] aliases",
+			"[the] permission [of [[the] command[s] %-strings%]]", "command[s] %strings%'[s] permission",
+			"[the] permission message [of [[the] command[s] %-strings%]]", "command[s] %strings%'[s] permission message",
+			"[the] plugin [owner] [of [[the] command[s] %-strings%]]", "command[s] %strings%'[s] plugin [owner]");
 	}
 
-	@SuppressWarnings("null")
-	InfoType type;
-	@SuppressWarnings("null")
-	Expression<String> commandName;
+	@SuppressWarnings("NotNullFieldNotInitialized")
+	private InfoType type;
+
+	@Nullable
+	private Expression<String> commandName;
 
 	@Override
+	@SuppressWarnings("unchecked")
 	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
 		commandName = (Expression<String>) exprs[0];
+		if (commandName == null && !getParser().isCurrentEvent(ScriptCommandEvent.class))
```

Comment:
Can we also support `PlayerCommandPreprocessEvent.class`, `ServerCommandEvent.class`? (on command)